packages/auth/amplify_auth_cognito_dart/lib/src/state/state/configuration_state.dart

```diff
@@ -3,6 +3,33 @@
 
 import 'package:amplify_auth_cognito_dart/src/state/state.dart';
 import 'package:amplify_core/amplify_core.dart';
+import 'package:amplify_secure_storage_dart/amplify_secure_storage_dart.dart';
+import 'package:meta/meta.dart';
+
+/// Token for storing the [SecureStorageInterface] used to store Pinpoint Endpoint Id.
+@immutable
+class AmplifySecureStorageToken extends Token<SecureStorageInterface> {
+  /// Create a token using [name].
+  const AmplifySecureStorageToken(this.name, [super.dependencies]);
+
+  /// The token name.  Used for comparisons.
+  final String name;
+
+  @override
+  bool operator ==(Object other) =>
+      other is AmplifySecureStorageToken && name == other.name;
+
+  @override
+  int get hashCode => name.hashCode;
+
+  @override
+  String toString() => name;
+}
+
+/// Secure Storage Token for Pinpoint Endpoint storage.
+AmplifySecureStorageToken analyticsStorageToken = AmplifySecureStorageToken(
+  AmplifyStorageScope.awsPinpointAnalyticsPlugin.name,
+);
```

Comment:
Not in love with this and doesn't seem to be used everywhere

Reply:
Outdated, removed for EndpointInfoStoreManager